### PR TITLE
Re added auto download java

### PR DESCRIPTION
**Pull request by Shawiizz**
Hey ! I re added auto download Java 8. The advantages of this :
- People can use other java version on their machine
- People who don't know how to change / install java 8 can use this easy option 
- People who don't want to use java 8 on their machine
- And a lot of other reasons...
I tested this and it's working good.

### CODE REVIEW

File: src/main/java/red/mohist/network/download/DownloadJava [hunk 1]
```diff
@@ -0,0 +1,105 @@
+package red.mohist.network.download;
```
Comment (ishland) [inline src/main/java/red/mohist/network/download/DownloadJava:1]: Please use proper file extension name

File: src/main/java/red/mohist/Mohist.java [hunk 2]
```diff
@@ -25,14 +26,9 @@ public static String getVersion() {
...
   public static void main(String[] args) throws Throwable {
-    if(Float.parseFloat(System.getProperty("java.class.version")) != 52.0) {
-      System.out.println(Message.getString("unsupported.java.version"));
-      System.exit(0);
-    }
-    if(System.getProperty("log4j.configurationFile") == null) {
-      System.setProperty("log4j.configurationFile", "log4j2_mohist.xml");
-    }
-      System.out.println("\n" +
+    if(Float.parseFloat(System.getProperty("java.class.version")) != 52.0 || bMohist("use_custom_java8", "false"))
```
Comment (ishland) [inline src/main/java/red/mohist/Mohist.java:29]:
```suggestion
    if(Float.parseFloat(System.getProperty("java.class.version")) != 52.0 || bMohist("use_custom_java8"))
```
There's no point of using a defaultReturn. It is already false.

File: src/main/java/red/mohist/configuration/MohistConfigUtil.java [hunk 3]
```diff
@@ -16,33 +16,37 @@
...
+    } catch (IOException e) {
+      return defaultreturn;
     }
+  }
 
-    public static boolean getBoolean(File f, String key) {
-        return !getString(f, key, "true").equals("false");
-    }
+  public static boolean getBoolean(File f, String key, String b) {
+    return !getString(f, key, b).equals("false");
```
Comment (ishland) [inline src/main/java/red/mohist/configuration/MohistConfigUtil.java:38]: What's the point of adding this?

File: src/main/java/red/mohist/configuration/MohistConfigUtil.java [hunk 4]
```diff
@@ -54,10 +58,16 @@ public static void copyMohistConfig() {
...
-      new MohistConfig().init();
+    new MohistConfig().init();
   }
 
-  public static boolean bMohist(String key) { return MohistConfigUtil.getBoolean(new File("mohist-config", "mohist.yml"), key+":"); }
+  public static boolean bMohist(String key) {
+    return MohistConfigUtil.getBoolean(new File("mohist-config", "mohist.yml"), key);
+  }
+
+  public static boolean bMohist(String key, String defaultReturn) {
```
Comment (ishland) [inline src/main/java/red/mohist/configuration/MohistConfigUtil.java:68]: Remove this as the defaultReturn is already false

**Comment by TansuoTro** [comment]
OK

File: src/main/java/red/mohist/configuration/MohistConfigUtil.java (hunk 4 above)
Comment (Shawiizz) [reply src/main/java/red/mohist/configuration/MohistConfigUtil.java:68]: No, it's true at default. It's why i made another function to put if it's true or false if the option doesn't exists in mohist.yml

Comment (ishland) [reply src/main/java/red/mohist/configuration/MohistConfigUtil.java:68]: I see. And it is true for me. Resolved

File: src/main/java/red/mohist/Mohist.java (hunk 2 above)
Comment (ishland) [reply src/main/java/red/mohist/Mohist.java:29]: Resolved

**Comment by Shawiizz** [comment]
As i can see i made something wrong because i edited the files on github directly.. The build failed

**Comment by hs-err** [comment]
apt install apt
yum install yum
apk add apk
pip install pip
java -jar Mohist.jar
............

**Pull request merged**